# Phase Rotation
### *Chapter 10 - Polyphase AC Circuits*

## Three-Phase Alternator

Let’s take the three-phase alternator design laid out earlier and watch what happens as the magnet rotates.

![three-phase-alternator](https://www.allaboutcircuits.com/uploads/articles/three-phase-alternator.jpg)

The phase angle shift of 120° is a function of the actual rotational angle shift of the three pairs of windings.

If the magnet is rotating clockwise, winding 3 will generate its peak instantaneous voltage exactly 120° (of alternator shaft rotation) after winding 2, which will hit its peak 120° after winding 1. The magnet passes by each pole pair at different positions in the rotational movement of the shaft.

Where we decide to place the windings will dictate the amount of phase shift between the windings’ AC voltage waveforms.

If we make winding 1 our “reference” voltage source for phase angle (0°), then winding 2 will have a phase angle of -120° (120° lagging, or 240° leading) and winding 3 an angle of -240° (or 120° leading).

## Phase Sequence

This sequence of phase shifts has a definite order. For clockwise rotation of the shaft, the order is 1-2-3 (winding 1 peak first, them winding 2, then winding 3). This order keeps repeating itself as long as we continue to rotate the alternator’s shaft.

![phase-sequence](https://www.allaboutcircuits.com/uploads/articles/clockwise-rotation-phase.jpg)

*Note that the above notion*, and anything that depends on it, including any plot, assumes that the positions of $2$b and $2$a are swapped in the above diagram and that positions of winding pairs $2$ and $3$ are subsequently swapped.  Without this assumption, winding $2$ would have a phase angle of $-60^\circ$ (i.e. it would lag by $60^\circ$) and winding $3$ would have a phase angle of $-120^\circ$ (i.e. it would lag by $120^\circ$).

Below is a diagram of an alternator with the correct winding configuration.  

![corrected-alternator](three-phase-alternator.svg)

Now, getting back to describing the phase sequence, when the rotation is reversed -- i.e., when the alternator's shaft is rotated in the counter-clockwise direction -- the phase sequence's order reverses accordingly.

![phase-sequence-reversed](https://www.allaboutcircuits.com/uploads/articles/counterclockwise-rotation-phase.jpg)

## Phase Sequence Detectors

Since voltmeters and ammeters would be useless in telling us what the phase rotation of an operating power system is, we need to have some other kind of instrument capable of doing the job.

One ingenious circuit design uses a capacitor to introduce a phase shift between voltage and current, which is then used to detect the sequence by way of comparison between the brightness of two indicator lamps in the figure below.

![phase-sequence-detector](https://www.allaboutcircuits.com/uploads/articles/phase-sequence-detector.jpg)

The two lamps are of equal filament resistance and wattage. The capacitor is sized to have approximately the same amount of reactance at system frequency as each lamp’s resistance.

If the capacitor were to be replaced by a resistor of equal value to the lamps’ resistance, the two lamps would glow at equal brightness, the circuit is balanced. However, the capacitor introduces a phase shift between voltage and current in the third leg of the circuit equal to 90°.

This phase shift, greater than 0° but less than 120°, skews the voltage and current values across the two lamps according to their phase shifts relative to phase 3.

### With Resistor

Herein is a full analysis with a resistor with the same resistance, and subsequently power dissapation or wattage, as the two lamps' filament resistance and wattage in place of capacitor connected to phase \#3.

Suppose all three loads have a resistance of $2650\,\Omega$ and are connected to a power supply (an alternator) with a voltage of $120 \mathrm{V}$ and phase angles, from \#$1$ to \#$3$, denoted as '$\theta_1$', '$\theta_2$', and '$\theta_3$' respectively, of $\theta_1 = 0$, $\theta_2 = 120^\circ$, and $\theta_3 = 240^\circ$.

The amplitude of each current is given by, via Ohm's Law,  

$$ |E| = R|I|$$

$$ |I| = \frac{|E|}{R} = \frac{120\mathrm{V}}{2650\,\Omega}\approxeq 45.283\,\mathrm{mA}$$

In [17]:
1000*120/2650

45.283018867924525

And the phase angle of each current, since the loads are balanced, is equal to the phase angle of the the voltage of it's respective load.

Now, let's **verify** this against KCL.  Via KCL,  

$$ I_1 + I_2 + I_3 = 0$$

$$ 45.283\,\mathrm{mA}\,\angle\,0 \:+ \:45.283\,\mathrm{mA}\,\angle\,120^\circ = - 45.283\,\mathrm{mA}\,\angle\,240^\circ = 45.283\,\mathrm{mA}\,\angle\,60^\circ$$

In rectangular form, the above can be rewritten as:  

$$ 45.283\,\mathrm{mA}\,\big(1 + \cos120^\circ + j\cdot\sin120^\circ \big) = 45.283\,\mathrm{mA}\big(\cos60^\circ + j\cdot\sin60^\circ \big)$$

Dividing throughout by the amplitude gives:  

$$ \big(1 + \cos120^\circ + j\cdot\sin120^\circ\big ) = \big(\cos60^\circ + j\cdot\sin60^\circ\big)$$

In [18]:
import numpy as np

In [19]:
(np.cos(np.radians(120)), np.sin(np.radians(120)))

(-0.4999999999999998, 0.8660254037844387)

In [20]:
(np.cos(np.radians(60)), np.sin(np.radians(60)))

(0.5000000000000001, 0.8660254037844386)

Note that, throughout this subsection, it is assumed that the polarity of the power system is the same as the one shown below, in the next subsection, with the added assumption that the positive poles of each load are adjacent to the positive poles of their respective power supply.

### With Capacitor

Suppose we have the following circuit:  

![phase-sequence-detector-w/capacitor](https://www.allaboutcircuits.com/uploads/articles/spice-circuit-for-phase-sequence-detector.jpg)

First, let's consider that the frequency ($f$) can be found via the presuppostion that the reactance of $\mathrm{C}_1$, denoted as '$X_{\mathrm{C}1}$', is equal to $2650\,\Omega$.  

$$ X_{\mathrm{C}1} = \frac{1}{2\pi f C_1} = 2650\,\Omega = R_1, R_2$$

$$ f = \frac{1}{2\pi \cdot 2650\,\Omega\cdot1\mu\mathrm{F}}\approxeq60\mathrm{Hz}$$

In [21]:
f = 1 / (2*np.pi*2650*1e-6)
f

60.05846909128126

In [22]:
def cos(deg: float):
    return np.cos(np.radians(deg))

def sin(deg: float):
    return np.sin(np.radians(deg))

First we will use mesh current analysis to find the two currents flowing through load\#$1$ and \#$2$, denoted as '$I_1$' and '$I_2$' respectively.  Via Ohm's Law, KVL, and substitution, assuming the top mesh current, $I_1$, flows in the clockwise direction while the bottom mesh current, $I_2$, flows in the counter-clockwise direction,  

\begin{align*}\tag{1}
    120\mathrm{V}\angle\,0^\circ &= R_1I_1 + Z_\mathrm{C}\,(I_1+I_2) + 120\mathrm{V}\angle\,240^\circ \\
     &= I_1\cdot2650\,\Omega\angle\,0^\circ \:+\: (I_1+I_2)\cdot2650\,\Omega\angle\!-90^\circ \:+\: 120\mathrm{V}\angle\,240^\circ
\end{align*}    

and,
\begin{align*}\tag{2}
    120\mathrm{V}\angle\,120^\circ &= R_2I_2 + Z_\mathrm{C}\,(I_1+I_2) + 120\mathrm{V}\angle\,240^\circ \\
     &= I_2\cdot 2650\,\Omega\angle\,0^\circ \:+\: (I_1+I_2)\cdot 2650\,\Omega\angle\!-90^\circ \:+\: 120\mathrm{V}\angle\,240^\circ
\end{align*}

First, we solve for $I_1$ in terms of $I_2$.  Via equation (1),  

$$ 120\mathrm{V}\,0^\circ = I_1\cdot 2650\,\Omega\,\big(1 - j \big) + 120\mathrm{V}\angle\,240^\circ + I_2\cdot2650\,\Omega\angle\!-90^\circ$$

$$ I_1\cdot 2650\,\Omega\,\big(1 - j \big) = 120\mathrm{V}\,0^\circ  \:-\: 120\mathrm{V}\angle\,240^\circ \:-\: I_2\cdot2650\,\Omega\angle\!-90^\circ$$

In [23]:
np.degrees(np.arctan(-1))

-45.0

\begin{align*}
    I_1\cdot 2650\sqrt{2}\,\Omega\,\angle\!-45^\circ &= 120\mathrm{V}\,0^\circ  \:-\: 120\mathrm{V}\angle\,240^\circ \:-\: I_2\cdot2650\,\Omega\angle\!-90^\circ \\
     &= 120\mathrm{V}\,\big( 1\angle\,0^\circ - 1\angle\,240^\circ \big)\:-\:I_2\cdot2650\,\Omega\angle\!-90^\circ \\
     &= 120\mathrm{V}\,\big( 1 - \cos240^\circ - j\cdot\sin240^\circ \big)\:-\:I_2\cdot2650\,\Omega\angle\!-90^\circ \\
     &\approxeq 120\mathrm{V}\,(1.5+0.866j)\:-\:I_2\cdot2650\,\Omega\angle\!-90^\circ \\
     &\approxeq 207.85\mathrm{V}\angle\,30^\circ \:-\:I_2\cdot2650\,\Omega\angle\!-90^\circ
\end{align*}

Thus, we have:  

$$ I_1\cdot 2650\sqrt{2}\,\Omega\,\angle\!-45^\circ \approxeq 207.85\mathrm{V}\angle\,30^\circ \:-\:I_2\cdot2650\,\Omega\,\angle\!-90^\circ$$

In [24]:
x = complex(1-cos(240), -sin(240))
x

(1.5000000000000004+0.8660254037844384j)

In [34]:
120*np.sqrt(x.real**2 + x.imag**2)

207.8460969082653

In [26]:
np.degrees(np.arctan(x.imag/x.real))

29.999999999999986

Dividing throughout by $2650\sqrt{2}\,\Omega\,\angle\!-45^\circ$ gives:  

\begin{align*}\tag{3}
    I_1 &\approxeq \frac{207.85\mathrm{V}}{2650\sqrt{2}\,\Omega}\,\angle\,75^\circ \:-\:\frac{I_2}{\sqrt{2}}\,\angle\!-45^\circ \\ \\
     &\approxeq 55.46\,\mathrm{mA}\,\angle\,75^\circ \:-\:\frac{I_2}{\sqrt{2}}\,\angle\!-45^\circ
\end{align*}

In [35]:
207.85/(2650*np.sqrt(2))

0.055461186592310904

We can now solve for $I_2$ in terms of known quantities.  Via equations (2) and (3) in conjunction with substitution:  

\begin{align*}
    120\mathrm{V}\angle\,120^\circ &\approxeq I_2\cdot 2650\,\Omega\,\angle\,0^\circ \:+\: \left( \frac{207.85\mathrm{V}}{2650\sqrt{2}\,\Omega}\,\angle\,75^\circ \:-\:\frac{I_2}{\sqrt{2}}\,\angle\!-45^\circ+I_2\right)\cdot 2650\,\Omega\,\angle\!-90^\circ \:+\: 120\mathrm{V}\angle\,240^\circ \\ \\
     &\approxeq  I_2\cdot 2650\,\Omega\,\angle\,0^\circ \:+\: \frac{207.85\mathrm{V}}{\sqrt{2}}\,\angle\!-15^\circ \:+\: \left(I_2 - \frac{I_2}{\sqrt{2}}\,\angle\!-45^\circ \right)\cdot 2650\,\Omega\,\angle\!-90^\circ \:+\: 120\mathrm{V}\angle\,240^\circ \\ \\
     &\approxeq I_2\cdot 2650\,\Omega\,\angle\,0^\circ \:+\: \frac{207.85\mathrm{V}}{\sqrt{2}}\,\angle\!-15^\circ \:+\:I_2\cdot 2650\,\Omega\angle\!-90^\circ \:-\: I_2\cdot \frac{2650\,\Omega}{\sqrt{2}}\,\angle\!-135^\circ \:+\: 120\mathrm{V}\angle\,240^\circ \\ \\
     &\approxeq I_2\cdot 2650\,\Omega\,\left(1 - j - \frac{1}{\sqrt{2}}\,\angle\!-135^\circ \right) \:+\: \frac{207.85\mathrm{V}}{\sqrt{2}}\,\angle\!-15^\circ \:+\: 120\mathrm{V}\angle\,240^\circ
\end{align*}     

In [28]:
x2 = complex(1 - cos(-135), -1-sin(-135))
x2

(1.7071067811865475-0.2928932188134524j)

In [29]:
y = complex(cos(-15) + cos(240), sin(-15)+sin(240))
y

(0.46592582628906787-1.1248444488869591j)

Thus we have:  

\begin{align*}
    I_2\cdot 2650\,\Omega\,(1.707 - 0.293j) &\approxeq 120\mathrm{V}\angle\,120^\circ - 120\mathrm{V}\,(0.466 - 1.125j) \\
     &\approxeq 120\mathrm{V}\,\big(1\angle\,120^\circ - 0.466 + 1.125j \big)\\
     &\approxeq 120\mathrm{V}\,(-0.966 + 1.991j)
\end{align*} 

In [30]:
y = complex(cos(120), sin(120)) - y
y

(-0.9659258262890676+1.990869852671398j)

In [31]:
theta_x = np.degrees(np.arctan(-0.293/1.707))
theta_y = np.degrees(np.arctan(-1.991/0.966))
(theta_x, theta_y)

(-9.739689263441297, -64.11811187705209)

and,
$$ I_2\cdot 2650\,\Omega\angle\!-9.74^\circ \approxeq 120\mathrm{V}\angle\!-64.12^\circ$$

In [32]:
64.12-9.74

54.38

Dividing throughout by $2650\,\Omega\angle\!-9.74^\circ$ gives:  

\begin{equation}\tag{4}
    I_2 \approxeq \frac{120\mathrm{V}}{2650\,\Omega}\,\angle\!-54.38^\circ\approxeq 45.28\,\mathrm{mA}\,\angle\!-54.38^\circ
\end{equation}    